nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -1378,12 +1391,12 @@ mod test {
         assert_eq!(silo_pools[0].0.id(), pool1_for_silo.id());
         assert_eq!(silo_pools[0].1.is_default, false);
 
-        // linking an already linked silo errors due to PK conflict
-        let err = datastore
+        // linking an already linked silo is fine
+        let new = datastore
```

Comment:
how is this working? Shouldn't this be causing a PK conflict on the `ip_pool_resource`, as it was before this PR?